tools/n64tool.c

```diff
@@ -160,21 +159,28 @@ int copy_file(FILE *dest, char *file, int byte_swap)
 
 int output_zeros(FILE *dest, int amount)
 {
-	if((amount & 3) != 0)
+	if(amount & 3 != 0)
 	{
 		/* Don't support odd word alignments */
 		return -1;
 	}
 	
+	if(amount <= 0)
+	{
+		/* We are done */
+		return 0;
+	}
+	
 	int i;
-	while (amount > 0) {
-		int sz = amount;
-		if (sz > sizeof(zero))
-			sz = sizeof(zero);
-		fwrite(zero, 1, sz, dest);
-		amount -= sz;
+	
+	for(i = 0; i < (amount >> 2); i++)

```

Comment:
This is reverting @rasky 's changes at https://github.com/DragonMinded/libdragon/commit/3c45c51277fab510daeacc4f336a07bb00713b10